From b45253c271183f4d23250b225d68069fcb8deeda Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Thu, 6 Nov 2014 18:14:55 +0100
Subject: [PATCH 1/1] kcl_acpi.c: do not use KCL_ACPI_Slot_No_Hotplug() on
 linux 3.17 or higher

---
 kcl_acpi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kcl_acpi.c b/kcl_acpi.c
index 6ed9d5e..5a950f2 100755
--- a/kcl_acpi.c
+++ b/kcl_acpi.c
@@ -831,6 +831,7 @@ KCL_ACPI_DevHandle ATI_API_CALL KCL_ACPI_GetAlternateHandle(KCL_ACPI_DevHandle p
 
 static acpi_status KCL_ACPI_Slot_No_Hotplug(KCL_ACPI_DevHandle handle, u32 lvl, void *data, void **rv)
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,17,0)
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,12,7)
    struct acpi_device *tdev;
    struct pci_dev *pdev = (struct pci_dev *)data;
@@ -846,6 +847,7 @@ static acpi_status KCL_ACPI_Slot_No_Hotplug(KCL_ACPI_DevHandle handle, u32 lvl,
       }
    }
 #endif
+#endif
    return 0;
 }
 
-- 
1.9.1

