diff --git a/firegl_public.c b/firegl_public.c
index 6d42297..2fcbc10 100755
--- a/firegl_public.c
+++ b/firegl_public.c
@@ -6520,7 +6520,11 @@ void ATI_API_CALL KCL_fpu_begin(void)
     /* The thread structure is changed with the commit below for kernel 3.3:
      * https://github.com/torvalds/linux/commit/7e16838d94b566a17b65231073d179bc04d590c8
      */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,2,0)
+    if (cur_task->thread.fpu.fpregs_active)
+#else
     if (cur_task->thread.fpu.has_fpu)
+#endif
 #else
     if (cur_task->thread.has_fpu)
 #endif
